cogs/channel_checker.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+import difflib
+
+import discord
+from discord.ext.commands import Bot, Cog
+
+from config import CONFIG
+
+
+def locate(channel_name, channel_list):
+    """Find the channel from id, and create location string ("between #abc and #def")"""
+    for i, c in enumerate(channel_list):
+        if c.name == channel_name:
+            if i == 0:
+                msg = f"before {channel_list[1].mention}"
+            if i == len(channel_list) - 1:
+                msg = f"after {channel_list[len(channel_list)-2].mention}"
+            else:
+                msg = (
+                    f"between {channel_list[i-1].mention} "
+                    f"and {channel_list[i+1].mention}"
+                )
+            return c, msg
+    return None, "somewhere"
+
+
+def channel_sort(channels):
+    """Sort text channels into actual order"""
+    channels = [c for c in channels if isinstance(c, discord.TextChannel)]
+    return sorted(channels, key=lambda c: c.position)
+
+
+async def channel_check(bot):
+    """Periodically checks for channels that have been reordered"""
+    await bot.wait_until_ready()
+
+    guild = bot.get_guild(CONFIG.UWCS_DISCORD_ID)
+    channel = bot.get_channel(CONFIG.UWCS_EXEC_SPAM_CHANNEL_ID)
+
+    previous = channel_sort(guild.channels)
+    while not bot.is_closed():
+        await asyncio.sleep(CONFIG.CHANNEL_CHECK_INTERVAL)
+
+        # Get channel ids for diff
+        current = channel_sort(guild.channels)
+        curr_channels = [c.name for c in current]
+        prev_channels = [c.name for c in previous]
+
+        if curr_channels == prev_channels:
+            continue
+
+        # Find and filter changes
+        changes = list(difflib.Differ().compare(prev_channels, curr_channels))
+
+        added = [c.strip("+ ") for c in changes if c[0] == "+"]
+        removed = [c.strip("- ") for c in changes if c[0] == "-"]
```

Comment:
Seems easier to use set operations here instead of relying on text names.
```py
added = set(current) - set(previous)
removed = set(previous) - set(current)
```

Reply:
This was my initial approach, but because of the shuffling and channel objects with the different positions are considered distinct. Seemed easier to find differences like this than write something bespoke

Reply:
Since they are just reordered, subtraction of sets isn't what we are looking for. It's more a sequence matching problem, which diff is particularly good for. It's annoying there is a bit of processing to reformat it, but the current still seems like the best solution to me.